js/main.js

```diff
@@ -290,22 +451,130 @@ xhr3.addEventListener('load', function () {
     var weaponContainer = document.createElement('div');
     weaponContainer.className = 'weapon-container';
 
+    var starButton = document.createElement('button');
+    starButton.className = 'star-button';
+
+    var isFavorite = favorites.some(function (favorite) {
+      return favorite.displayName === weaponName;
+    });
+
+    if (isFavorite) {
+      starButton.classList.add('favorite');
+      weaponContainer.style.order = -1;
+    }
+
+    starButton.addEventListener('click', function () {
+      var favoriteIndex = favorites.findIndex(function (favorite) {
+        return favorite.displayName === weaponName;
+      });
+
+      if (favoriteIndex === -1) {
+        favorites.unshift(weapon);
+        this.classList.add('favorite');
+        weaponContainer.style.order = -1;
+      } else {
+        favorites.splice(favoriteIndex, 1);
+        this.classList.remove('favorite');
+        weaponContainer.style.order = '';
+      }
+
+      localStorage.setItem('favorites', JSON.stringify(favorites));
```

Comment:
You shouldn't be saving to localStorage here, you should be saving to your data model, and then that would be saved to localStorage in `data.js`